### PR TITLE
Add KEDA to list of platform components

### PR DESCRIPTION
**Pull request by niklasbeinghaus**
- Infrastructure services
- Messaging and event services

Issue: #378

### CODE REVIEW

File: platforms-whitepaper/latest/index.md [hunk 1]
```diff
@@ -466,7 +466,7 @@ relating it to existing CNCF or CDF projects.
   <tr>
     <td>Infrastructure services</td>
     <td>Run application code, connect application components and persist data for applications</td>
-    <td>Kubernetes, Kubevirt, Knative, WasmEdge<br />CNI, Istio, Cilium, Envoy, Linkerd, CoreDNS<br />Rook, Longhorn, Etcd</td>
+    <td>Kubernetes, Kubevirt, Knative, WasmEdge, Keda<br />CNI, Istio, Cilium, Envoy, Linkerd, CoreDNS<br />Rook, Longhorn, Etcd</td>
```
Comment (tomkerkhove) [inline platforms-whitepaper/latest/index.md:469]:
```suggestion
    <td>Kubernetes, Kubevirt, Knative, WasmEdge, KEDA<br />CNI, Istio, Cilium, Envoy, Linkerd, CoreDNS<br />Rook, Longhorn, Etcd</td>
```

File: platforms-whitepaper/latest/index.md [hunk 2]
```diff
@@ -476,7 +476,7 @@ relating it to existing CNCF or CDF projects.
   <tr>
     <td>Messaging and event services</td>
     <td>Enable applications to communicate with each other asynchronously</td>
-    <td>Strimzi, NATS, gRPC, Knative, Dapr</td>
+    <td>Strimzi, NATS, gRPC, Knative, Dapr, Keda</td>
```
Comment (tomkerkhove) [inline platforms-whitepaper/latest/index.md:479]:
```suggestion
    <td>Strimzi, NATS, gRPC, Knative, Dapr, KEDA</td>
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Sorry for the massive nits but we have to follow KEDA's branding guidelines - https://github.com/kedacore/governance/blob/main/BRANDING.md

**Comment by tomkerkhove** [comment]
Sorry to be a pain, but wouldn't it be nice if we put autoscaling front and center by introducing a new category for it? Scalability is at the heart of cloud-native so I believe we should emphasize this here.

Not only to cover KEDA, but potentially other things such as Kubernetes Cluster Autoscaler or Karpenter.

Also, can we update this visual overview?

image

**Comment by niklasbeinghaus** [comment]
> Sorry to be a pain, but wouldn't it be nice if we put autoscaling front and center by introducing a new category for it? Scalability is at the heart of cloud-native so I believe we should emphasize this here.

I wasn't quite sure about the categories in this either, however from a platform engineering perspective, it makes sense to keep this condensed. What do you think about splitting the "automate" category and adding a "scaling" category thereunder?

> Also, can we update this visual overview?
Once we've found a solution for the right category, we should definitely update the visual overview as well.

**Comment by tomkerkhove** [comment]
> I wasn't quite sure about the categories in this either, however from a platform engineering perspective, it makes sense to keep this condensed. What do you think about splitting the "automate" category and adding a "scaling" category thereunder?

Fine for me!

**Comment by joshgav** [comment]
Hi @tomkerkhove, I lean to think (auto)scaling is a feature of whatever resource is being managed. Since KEDA scales compute infrastructure based on events it seems it's a feature of compute.

Karpenter and Cluster Autoscaler are interesting examples. Still, they're mostly focused on automatically scaling compute infrastructure and so probably belong with "Infrastructure". Do they also automate scaling of other resource types?

So I'd say for now let's put autoscalers together with the resource types they scale, in KEDA's case infrastructure/compute.

**Review by joshgav — CHANGES_REQUESTED** [review]
We've tried to list any given project only once to reduce clutter. Though it fits a bit under "events" too, I think the primary place for KEDA would be within infrastructure since it manages compute. Could you put it just in the infrastructure category? Thanks!

**Comment by zroubalik** [comment]
> Hi @tomkerkhove, I lean to think (auto)scaling is a feature of whatever resource is being managed. Since KEDA scales compute infrastructure based on events it seems it's a feature of compute.
> 

Hey, it depends on what do you consider as `compute infrastructure`, KEDA scales workloads (apps - Deployments, Jobs...).

> Karpenter and Cluster Autoscaler are interesting examples. Still, they're mostly focused on automatically scaling compute infrastructure and so probably belong with "Infrastructure". Do they also automate scaling of other resource types?

These tools scales only compute nodes (k8s nodes) and if I am not mistaken Karpenter is AWS only.

**Comment by abangser** [comment]
I would agree that this still feels like compute scaling. IMO deployments / jobs / CRDs are versions of compute, just like bare metal, VMs, and containers are versions of compute. We are getting better at abstractions, but at the end of the day my understanding is that KEDA helps manage higher load on the system by horizontally scaling the things that respond to those higher requests. Would be really helpful to understand if that is the wrong understanding though!

**Comment by zroubalik** [comment]
> I would agree that this still feels like compute scaling. IMO deployments / jobs / CRDs are versions of compute, just like bare metal, VMs, and containers are versions of compute. We are getting better at abstractions, but at the end of the day my understanding is that KEDA helps manage higher load on the system by horizontally scaling the things that respond to those higher requests. Would be really helpful to understand if that is the wrong understanding though!

Yeah, that's completely correct understanding! As I mentioned, it depends on what do you see by `compute`, if it is what you just mentioned above, then we should be good.

**Review by joshgav — APPROVED** [review]
LGTM! Thanks for the productive discussion!

**Pull request merged**